### PR TITLE
feat(csv): support setting escape.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

default `""` to escape  `"`
add support `\"`

`FILE_FORMAT = (type = 'CSV' escape='\\')"` for copy

`-H "format_escape:'\\'` for streaming load

### CODE REVIEW

**Review by ClSlaid — APPROVED** [review]

**Review by drmingdrmer — COMMENTED** [review]
Add another `VER` line here:
https://github.com/datafuselabs/databend/blob/421c0fb60c26c7a3ce551bcec1a579f9c4fee753/src/meta/proto-conv/src/util.rs#L65

Add a test that decodes an encoded message of this version, just like:
https://github.com/datafuselabs/databend/blob/bb793d88e03e2acb75ef5889385b9e45721cef92/src/meta/proto-conv/tests/it/user_stage.rs#L55-L96

**Comment by youngsofun** [comment]
@drmingdrmer  only  add `user_stage_fs_v17`? I also see `test_user_stage_s3_v16`

**Comment by wubx** [comment]
```
 cat t.csv
"id","name","upadte_at","content"
1,"a","2022-10-25 10:51:14","{\"hello\":\"world\"}"
2,"b","2022-10-25 10:51:28","{\"金角大王\":\"沙河\"}"
3,"c","2022-10-25 10:51:50","{\"银角大王\":\"沙河\"}"

MySQL [(none)]> copy into tb1 from @my_stage files=('t.csv') file_format=(type='CSV' skip_header=1 escape='\\');
ERROR 1105 (HY000): Code: 1046, displayText = fail to parse CSV t.csv:1 expect 4 fields, only found 1  .

```

streaming_load work ok.

**Comment by youngsofun** [comment]
>>> Thus you have to use the next version: VER=18.
ok

File: src/meta/proto-conv/src/util.rs
Comment (drmingdrmer) [inline src/meta/proto-conv/src/util.rs:66]: This will conflict with another PR:
- #8526

**Comment by youngsofun** [comment]
> ```
>  cat t.csv
> "id","name","upadte_at","content"
> 1,"a","2022-10-25 10:51:14","{\"hello\":\"world\"}"
> 2,"b","2022-10-25 10:51:28","{\"金角大王\":\"沙河\"}"
> 3,"c","2022-10-25 10:51:50","{\"银角大王\":\"沙河\"}"
> 
> MySQL [(none)]> copy into tb1 from @my_stage files=('t.csv') file_format=(type='CSV' skip_header=1 escape='\\');
> ERROR 1105 (HY000): Code: 1046, displayText = fail to parse CSV t.csv:1 expect 4 fields, only found 1  .
> ```
> 
> streaming_load work ok.

should specify file_format when crate stage?

File: src/meta/proto-conv/src/util.rs
Comment (youngsofun) [reply src/meta/proto-conv/src/util.rs]: changed to 18.

**Comment by youngsofun** [comment]
copy from s3 location works well

**Comment by wubx** [comment]
> copy from s3 location works well

Here looks ci ok, let's merge it.

**Review by wubx — APPROVED** [review]

**Review by drmingdrmer — APPROVED** [review]
It looks good to me but I'm afraid it has to wait until #8526 being merged:
- #8526

**Comment by youngsofun** [comment]
test_sqllogic_base_cluster_linux  timeout after  30m

File: src/common/io/src/format_settings.rs
Comment (youngsofun) [inline src/common/io/src/format_settings.rs:16]: wait until https://github.com/datafuselabs/databend/pull/8526 being merged:

https://github.com/datafuselabs/databend/pull/8526

Comment (youngsofun) [reply src/common/io/src/format_settings.rs:16]: cc @drmingdrmer

**Review by sundy-li — APPROVED** [review]

**Pull request merged**
